lib/elixir/lib/record.ex

```diff
@@ -413,17 +418,26 @@ defmodule Record do
   def __keyword__(atom, fields, record) do
     if is_record(record, atom) do
       [_tag | values] = Tuple.to_list(record)
-      join_keyword(fields, values, [])
+      case join_keyword(fields, values, []) do
+        kv when is_list(kv) ->
+          kv
+        expected_fields ->
+          msg = "expected argument can be a #{inspect atom} record with #{expected_fields} fields, while got: #{inspect record}"
```

Comment:
Can we improve the message to:

    "expected argument to be a #{inspect atom} record with #{expected_fields} fields, got: #{inspect record}"

Thank you!